geode-core/src/main/java/org/apache/geode/internal/tcp/Connection.java

```diff
@@ -2527,7 +2527,7 @@ private void writeAsync(SocketChannel channel, ByteBuffer buffer, boolean forceA
         long queueTimeoutTarget = now + asyncQueueTimeout;
         channel.configureBlocking(false);
         try {
-          try (final ByteBufferSharing outputSharing = ioFilter.wrap(buffer)) {
+          try (final ByteBufferSharing outputSharing = ioFilter.wrap(buffer, channel)) {
```

Comment:
I recommend extracting some methods throughout the code you're touching to minimize nested try-blocks, if-blocks and loops. It really helps clean it up and improve readability.

Reply:
Thanks for pointing this out @kirklund. I'd like to do a minimal touch to product code in this PR so it is easily back-ported. I will make a note to clean up nested trys in Connection in a future PR.